x/auth/keeper/deterministic_test.go

```diff
@@ -0,0 +1,122 @@
+
+	maccPerms := map[string][]string{
+		"fee_collector":          nil,
+		"mint":                   {"minter"},
+		"bonded_tokens_pool":     {"burner", "staking"},
+		"not_bonded_tokens_pool": {"burner", "staking"},
+		multiPerm:                {"burner", "minter", "staking"},
+		randomPerm:               {"random"},
+	}
+
+	suite.accountKeeper = keeper.NewAccountKeeper(
+		suite.encCfg.Codec,
+		key,
+		types.ProtoBaseAccount,
+		maccPerms,
+		"cosmos",
+		types.NewModuleAddress("gov").String(),
+	)
+
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.encCfg.InterfaceRegistry)
+	types.RegisterQueryServer(queryHelper, suite.accountKeeper)
+	suite.queryClient = types.NewQueryClient(queryHelper)
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryAccounts() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		addr := testdata.AddrTestDeterministic(t)
+		suite.accountKeeper.SetAccount(suite.ctx,
+			suite.accountKeeper.NewAccountWithAddress(suite.ctx, addr))
+
+		acc, err := suite.queryClient.Account(suite.ctx, &types.QueryAccountRequest{Address: addr.String()})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(acc)
+
+		var prevRes types.AccountI
+		err = suite.encCfg.InterfaceRegistry.UnpackAny(acc.Account, &prevRes)
+		suite.Require().NoError(err)
+
+		for i := 0; i < 1000; i++ {
+			acc, err := suite.queryClient.Account(suite.ctx, &types.QueryAccountRequest{Address: addr.String()})
+			suite.Require().NoError(err)
+			suite.Require().NotNil(acc)
+			var account types.AccountI
+
+			err = suite.encCfg.InterfaceRegistry.UnpackAny(acc.Account, &account)
+			suite.Require().NoError(err)
+			suite.Require().Equal(account.GetAddress(), addr)
+
+			// check with previous response too.
+			suite.Require().Equal(account.GetAddress(), prevRes.GetAddress())
+			suite.Require().Equal(account.GetPubKey(), prevRes.GetPubKey())
+			suite.Require().Equal(account.GetSequence(), prevRes.GetSequence())
+			suite.Require().Equal(account.GetAccountNumber(), prevRes.GetAccountNumber())
+			prevRes = account
+		}
+	})
+
+	priv := secp256k1.GenPrivKey()
+	pub := priv.PubKey()
+	addr1 := sdk.AccAddress(priv.PubKey().Address())
+	// randAccNumber :=
+	accNum := uint64(rand.Intn(100) + 10000) // range 10000 to 10100
+	seq := uint64(0)
+
+	acc1 := types.NewBaseAccount(addr1, pub, accNum, seq)
+	suite.accountKeeper.SetAccount(suite.ctx, acc1)
+
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		acc, err := suite.queryClient.Account(suite.ctx, &types.QueryAccountRequest{Address: addr1.String()})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(acc)
+		var account types.AccountI
+
+		err = suite.encCfg.InterfaceRegistry.UnpackAny(acc.Account, &account)
+		suite.Require().NoError(err)
+
+		suite.Require().Equal(account.GetAddress(), addr1)
+		suite.Require().Equal(account.GetPubKey(), pub)
+		suite.Require().Equal(account.GetAccountNumber(), accNum)
+		suite.Require().Equal(account.GetSequence(), seq)
+	})
```

Comment:
With the hardcoded address test, we should not be using rapid, but do the `for i = 1..1000` iteration.

Maybe we can extract the for loop in a function, the test looks like:
```
// test 1000 iterations with rapid:
rapid.Check(func(){
  addr := AddressGenerator().Draw()
  runIterations(addr)
})


// then test regression
addr := <hardcoded>
runIterations(addr)
```